### ISSUE TITLE
Change the structure of Bumblebee in order to easily build and maintain packages

### ISSUE THREAD

**Issue by glyptostroboides** [MrMEEE/bumblebee-Old-and-abbandoned#237]
Hello all,

I'm trying to build a package for Ubuntu. I have already review the code to see how to get it into a package. But they are many changes, we will have to perform in order to package Bumblebee easily. And they are also idea that must be kept or improved.

-Package installation (Virtual-GL, Nvidia drivers, screen ??) must be done first in the script because all existing package will be dependencies of bumblebee packages and should be installed before. Acpi_call should also be turned into is own package and be installed as a dependencie.

-All the copy file event must be gathered in the same script to be done at the same time (which will not be need in the packager because it install the files by copying the file system that is in the package), and  

-The command lines which build a file must be converted to a file already preset in the package file hierarchy.

-All command line which are supposed to configure Bumblebee must be gathered in a separate script which should be run by the package installer after he has copied the files but should also be run to reconfigure Bumblebee (we have to merge the configuration stages from install.sh and the bumblebee-config script in order to get a simple structure). 

-This configuration settings should also be stored in a file that could be keep when upgrading package. It would also be nice to have a per user configuration stored in home directory.

-All backup event must be gathered in the same script (there still some backup event that are performed in other scripts than the backupconfiguration script).

-User right setting should be gathered in the same script in order to easily manage the rights using the package way.

-For uninstall process, the stages should be also gathered by this way.

-Some settings should be used as arguments of optirun script in order to easily build per application configuration settings.

So I've already build a package but haven't tried it : i'm lacking a testing environnement. It is also hand build with an old version only for Ubuntu 11.04 -64bits. But bumblebee change so fast and in order to maintain a package for Ubuntu change in the structure of bumblebee install process will help to keep this package up to date and to easily manage upgrades in a clean way.

Sorry for my english.
Davy.
